bin/pygrb/pycbc_pygrb_plot_coh_ifosnr

```diff
@@ -145,14 +141,12 @@ def plot_deviation(percentile, snr_grid, y, ax, style):
     # original code line (commented out above)
     y_vals = scipy.stats.ncx2.ppf(percentile, 2, y * y) ** 0.5
     y_vals = numpy.unique(y_vals)
-    x_vals = snr_grid[0 : len(y_vals)]
+    x_vals = snr_grid[0:len(y_vals)]
     n_vals = int(len(y_vals) / 2)
-    f = scipy.interpolate.interp1d(
-        x_vals[0:n_vals],
-        y_vals[0:n_vals],
-        bounds_error=False,
-        fill_value="extrapolate",
-    )
+    f = scipy.interpolate.interp1d(x_vals[0:n_vals],
+                                   y_vals[0:n_vals],
+                                   bounds_error=False,
+                                   fill_value="extrapolate")
```

Comment:
Since `interp1d` is deprecated we should replace it with either
```
y_vals_new = numpy.interp(snr_grid, x_vals[0:n_vals],
        y_vals[0:n_vals])
```
or
```
f = scipy.interpolate.CubicSpline(x_vals[0:n_vals],
                                   y_vals[0:n_vals])
```

Reply:
Which plot does this script make? I do not understand why interpolation is needed here (the saturation argument in the comment).

Reply:
It makes the lines for the single IFO vs coherent SNR [plots](https://ldas-jobs.ligo.caltech.edu/~francesco.pannarale/LVC/pygrb_may2024_test9/2._signal_consistency/2.02_individual_detector_snrs/).